stl/inc/ranges

```diff
@@ -4748,6 +4748,378 @@ namespace ranges {
+        range_difference_t<_Vw> _Count = 0;
+
+        template <class>
+        class _Iter_base {};
+
+        template <_Slide_caches_first _Base>
+        class _Iter_base<_Base> {
+        protected:
+            using _Base_iter = iterator_t<_Base>;
+            _Base_iter _Last_element{};
+
+        public:
+            _Iter_base() = default;
+            constexpr explicit _Iter_base(_Base_iter _Last_element_) noexcept(
+                is_nothrow_move_constructible_v<_Base_iter>)
+                : _Last_element{_STD move(_Last_element_)} {}
+
+
+            _NODISCARD constexpr const _Base_iter& _Get_last_element() const noexcept //
+                requires _Slide_caches_first<_Base> {
+                return _Last_element;
+            }
+        };
+
+        template <bool _Const>
+        class _Iterator : public _Iter_base<_Maybe_const<_Const, _Vw>> {
+        private:
+            friend _Iterator;
+            friend slide_view;
+
+            using _BaseTy    = _Iter_base<_Maybe_const<_Const, _Vw>>;
+            using _Base      = _Maybe_const<_Const, _Vw>;
+            using _Base_iter = iterator_t<_Base>;
+
+            /* [[no_unique_address]] */ _Base_iter _Current{};
+            range_difference_t<_Base> _Count = 0;
+
+            // clang-format off
+            constexpr _Iterator(_Base_iter _Current_, range_difference_t<_Base> _Count_) noexcept(
+                is_nothrow_move_constructible_v<_Base_iter>&&
+                    is_nothrow_default_constructible_v<_Base_iter>) /* strengthened */
+                requires (!_Slide_caches_first<_Base>)
+                : _Current(_STD move(_Current_)), _Count(_Count_) {}
+            // clang-format on
+
+            constexpr _Iterator(_Base_iter _Current_, _Base_iter _Last_, range_difference_t<_Base> _Count_) noexcept(
+                is_nothrow_move_constructible_v<_Base_iter>&&
+                    is_nothrow_default_constructible_v<_Base_iter>) /* strengthened */
+                requires _Slide_caches_first<_Base>
+                : _BaseTy(_STD move(_Last_)), _Current(_STD move(_Current_)), _Count(_Count_) {}
+
+        public:
+            using value_type        = decltype(_RANGES views::counted(_Current, _Count));
+            using difference_type   = range_difference_t<_Base>;
+            using iterator_category = input_iterator_tag;
+            using iterator_concept  = conditional_t<random_access_range<_Base>, random_access_iterator_tag,
+                conditional_t<bidirectional_range<_Base>, bidirectional_iterator_tag, forward_iterator_tag>>;
+
+            _Iterator() = default;
+
+            constexpr _Iterator(_Iterator<!_Const> _Other) noexcept(is_nothrow_move_constructible_v<_Base_iter>) //
+                requires _Const && convertible_to<iterator_t<_Vw>, _Base_iter> /* strengthened */
+                : _Current(_STD move(_Other._Current)), _Count(_Other._Count) {}
+
+            _NODISCARD constexpr value_type operator*() const

```

Comment:
note: the return type of this function is `auto` in the proposal, but I prefer spelling the type out explicitly and it shouldn't result in any problems; no changes requested.